### PR TITLE
nix: Add postgrest-release tool

### PR DESCRIPTION
**Pull request by wolfgangwalther**
As proposed in https://github.com/PostgREST/postgrest/issues/2006#issuecomment-1007661689.

I just created a new pre-release with this tool.

The output was the following:

```
nix-shell% postgrest-release
Current version is 9.0.0.20220107.
1) 9.0.0.20220130
2) 9.0.1
3) 9.1.0
4) 10.0.0
Please select the new version: 1
Selected 9.0.0.20220130 ...
Updating postgrest.cabal ...
Committing ...
[...]
Tagging ...
To push both the branch and the new tag, the following will be run:

git push --atomic upstream main v9.0.0.20220130

Proceed? (y/N) y
[...]
To github.com:PostgREST/postgrest.git
   58f76f3d..c894b543  main -> main
 * [new tag]           v9.0.0.20220130 -> v9.0.0.20220130
```

This can be used for both regular releases and pre-releases.

### CODE REVIEW

**Comment by wolfgangwalther** [comment]
@steve-chavez would anything be blocking us from releasing a `9.1.0` right now? We have quite a few bugfixes lined up already.

File: nix/tools/release/default.nix [hunk 1]
```diff
@@ -46,9 +46,82 @@ let
...
+        git commit -m "bump version to $new_version" > /dev/null
+
+        echo "Tagging ..."
+        git tag "v$new_version" > /dev/null
+
+        trap "Couldn't find remote. Please push manually ..." ERR
+        remote="$(git remote -v | grep PostgREST/postgrest | grep push | cut -f1)"
+        trap "" ERR
+
+        push="git push --atomic $remote main v$new_version"
```
Comment (steve-chavez) [inline nix/tools/release/default.nix:104]: Cool. TIL about `atomic`. 

https://github.blog/2015-04-30-git-2-4-atomic-pushes-push-to-deploy-and-more/

**Review by steve-chavez — APPROVED** [review]

**Comment by steve-chavez** [comment]
> would anything be blocking us from releasing a 9.1.0 right now?

Ah, the docs for the admin server live/ready endpoints are pending, I see no blockers after those are done.

File: nix/tools/release/default.nix (hunk 1 above)
Comment (wolfgangwalther) [reply nix/tools/release/default.nix:104]: me2 :D

**Comment by steve-chavez** [comment]
Just tried it and it works flawlessly :100:

**Pull request merged**